virtualizarr/writers/icechunk.py

```diff
@@ -118,6 +117,75 @@ def dataset_to_icechunk(
     )
 
 
+def datatree_to_icechunk(
+    dt: xr.DataTree,
+    store: "IcechunkStore",
+    *,
+    last_updated_at: datetime | None = None,
+) -> None:
+    """
+    Write an xarray dataset to an Icechunk store.
+
+    Both `icechunk` and `zarr` (v3) must be installed.
+
+    Parameters
+    ----------
+    dt: xr.DataTree
+        DataTree to write to an Icechunk store. All variables must be backed by
+        ManifestArray objects.
+    store: IcechunkStore
+        Store to write the dataset to, which must not be read-only.
+    last_updated_at: datetime, optional
+        The time at which the virtual dataset was last updated. When specified, if any
+        of the virtual chunks written in this session are modified in storage after this
+        time, icechunk will raise an error at runtime when trying to read the virtual
+        chunk. When not specified, icechunk will not check for modifications to the
+        virtual chunks at runtime.
+
+    Raises
+    ------
+    ValueError
+        If the store is read-only.
+    """
+    try:
+        from icechunk import IcechunkStore  # type: ignore[import-not-found]
+        from zarr import Group  # type: ignore[import-untyped]
+        from zarr.storage import StorePath  # type: ignore[import-untyped]
+    except ImportError:
+        raise ImportError(
+            "The 'icechunk' and 'zarr' version 3 libraries are required to use this function"
+        ) from None
+
+    if not isinstance(store, IcechunkStore):
+        raise TypeError(
+            f"store: expected type IcechunkStore, but got type {type(store)}"
+        )
+
+    if not isinstance(last_updated_at, (type(None), datetime)):
+        raise TypeError(
+            "last_updated_at: expected type datetime,"
+            f" but got type {type(last_updated_at)}"
+        )
+
+    if store.read_only:
+        raise ValueError("supplied store is read-only")
+
+    for path, subtree in dt.subtree_with_keys:
+        tree = cast(xr.DataTree, subtree)  # subtree is typed as Unknown
+        ds = tree.to_dataset()
```

Comment:
There's some important subtleties here with inherited coordinate variables. To get the same behaviour as  `DataTree.to_zarr()` you need to pass the `inherit` kwargs, which should be exposed publicly by this accessor method as `write_inherited_coords`. See 

https://github.com/pydata/xarray/blob/08fa7b9fedaa977eed6ef4de1dbd8232b8b1ceed/xarray/core/datatree_io.py#L124

Reply:
Ah, thanks for the code reference. I'll adjust accordingly.